src/platform_impl/linux/wayland/seat/keyboard/mod.rs

```diff
@@ -32,7 +32,13 @@ impl Dispatch<WlKeyboard, KeyboardData, WinitState> for WinitState {
         _: &QueueHandle<WinitState>,
     ) {
         let seat_state = match state.seats.get_mut(&data.seat.id()) {
-            Some(seat_state) => seat_state,
+            Some(seat_state) => {
+                if seat_state.keyboard_state.is_none() {
+                    warn!("Received keyboard event {event:?} without keyboard");
+                    return;
+                }
+                seat_state
+            },
             None => return,
```

Comment:
Should we also warn on this branch?

Reply:
I've made it a bit more broken compositor proof, though, keep in mind that the old code is like that for like ~7 years, it's just some 4chan devs decided to yolo recently.